fms/modules/speculator.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from fms.modules.layernorm import LayerNormParameterized
+
+
+class Speculator(nn.Module):
+    def __init__(self, emb_dim=4096, vocab_size=32000, n_heads=4):
+        super().__init__()
+        self.nheads = n_heads
+        self.emb_dim = emb_dim
+        self.vsize = vocab_size
+        self.emb = nn.ModuleList([nn.Embedding(vocab_size, emb_dim) for _ in range(n_heads)])
+        self.proj = nn.ModuleList([nn.Linear(emb_dim * 2, emb_dim, bias=False) for _ in range(n_heads)])
+        self.head = nn.ModuleList([nn.Linear(emb_dim, vocab_size, bias=False) for _ in range(n_heads)])
```

Comment:
out of curiosity, how did you pick the architecture for this?

Reply:
Simplest and flattest recurrent MLP architecture I could come up with!